### DISCUSSION TITLE
Build HL7 message from scratch - not the result expected

### DISCUSSION THREAD

**Discussion by Migas1979** [nextgenhealthcare/connect#4564] [Q&A] [answered]
I would like to create a channel that detects a pdf file with a filename *.pdf.
The filename is always PatientIdentication_BED.pdf

I would like to build an hl7 message from scratch where the PatientIdentification goes to the PID segment. And The bed goes to PV1. And the pdf file is encapsulated in ORU.
the fileareader is ready but I don't know which data type should I insert.
I have the split part working well.
But as the end something is wrong because the output message is not what I pretend.
Any clues?

I am used to receive hl7 messages and transform them. Not to create from scratch.

**Comment by Migas1979** [comment, marked as answer]
This is my final message
MSH|^~\&|App||NUR||20210522200||ORU^R01|10072020530100004593|P|2.5||||AL|ER

Even after transforming and had adding
filename=sourceMap.get('originalFilename');
var parts= filename.split('_');
var pid=parts[0];
var bed=parts[1];

msg['PID']['PID.2']['PID.2.1']='1'
msg['PID']['PID.3']['PID.3.1']=pid;

msg['PV1']['PV1.2']['PV1.2.1']='I';
msg['PV1']['PV1.3']=bed;

**Reply by Hueli** [reply]
Where does the MSH come from? If it's in the template section, replace msg[] with tmp[] in your code. As soon as there's just one single character in the template section, tmp contains your message data.

**Reply by Migas1979** [reply]
Many thanks.
